### DISCUSSION TITLE
Building a custom email confirmation flow in React Native is not possible with the session data returned when email confirmations are toggled off.

### DISCUSSION THREAD

**Discussion by mypalvikram** [supabase/supabase#9122] [Questions] [answered]
I'll try my best to explain my use case. I'm building in React Native (not expo) for more context.

Currently, as it stands, if you have ```Confirm email``` toggled on and a user registers it sends the user an email to confirm. Until they do that, anytime they try to log in it will throw an error. 

Ok fine, Supabase is handling unverified emails for me. But, what if I want to handle the confirmation flow myself?

The user experience right now is signing up -> confirming email -> logging back in.

I don't want my users to have to enter their information twice. So my ideal user experience is when they sign up by email they get sent to an email confirmation page which lets them know the next steps. In the background, I refresh the user periodically to see if they confirmed their email so I can automatically navigate them (or they can resend with a button on this page). But I need a session in order to do that.

So from here, I try and build my custom user experience. I go to the dashboard and disable email confirmations so I can implement my custom flow. The problem now is when a user signs up, their ```email_confirmed_at``` is populated automatically.

I wish this wasn't the case and there was a field that could be used to manage email confirmations on my own. 

**Solution?:** Ideally even just a ```email_confirmed: boolean``` field that stayed false even if confirmations were disabled. This way I could navigate users to the page and send emails on my own.

If I'm missing something on how to implement my custom flow, please let me know. But after inspecting the data returned from both users after confirming and with confirmations off, I don't see a way to do this.

**Comment by kangmingtay** [comment, marked as answer]
Hey @MyPalVikram, thanks for the feedback! There isn't a clean way to do this currently besides query the `auth.users.email_confirmed_at` column in the db. However, this is something we'll definitely consider in the next major version of supabase auth since it makes the API less opinionated to allow developers to customise the auth flow as to how they want it.
